Sources/PubNub/KMP/KMPPubNub+Listeners.swift

```diff
@@ -28,45 +28,69 @@ extension KMPPubNub {
       onFileEvent: { [weak pubnub] in listener.onFile?(KMPFileChangeEvent.from(event: $0, with: pubnub)) },
       onAppContext: { listener.onAppContext?(KMPAppContextEventResult.from(event: $0)) }
     )
+
+    // Establishes an association to infer which Swift listener
+    // corresponds to which KMP listener
+    listener.underlying = swiftListener
+
+    return swiftListener
   }
 
   func createStatusListener(from listener: KMPStatusListener) -> StatusListener {
-    StatusListener(onConnectionStateChange: { [weak pubnub] newStatus in
+    let swiftStatusListener = StatusListener(onConnectionStateChange: { [weak pubnub] newStatus in
       guard let pubnub = pubnub else {
         return
       }
-      
+
       let category: KMPConnectionStatusCategory
       let errorIfAny: Error?
-      
+      let affectedChannels: [String]
+      let affectedGroups: [String]
```

Comment:
GitHub doesn't allow me to suggest changes for the whole piece. 
I was suggesting to have following by default and only for `case let .subscriptionChanged(channels, groups)` assigned to the values from status.

```suggestion
      let affectedChannels = pubnub.subscribedChannels
      let affectedGroups = pubnub.subscribedChannelGroups
```
For other cases except `.subscriptionChanged(channels, groups)` remove assignment.

Reply:
Gotcha, I'll fix it that way. 
There's one caveat, they have to be `var` to reassing them later for `.subscriptionChanged` 🙂

Reply:
Fixed!